### ISSUE TITLE
Properly handle task limit

### ISSUE THREAD

**Issue by whilefoo** [ubiquity-os-marketplace/command-start-stop#28]
The original implementation allowed for productive contributors to work on two tasks concurrently. With a starting two task limit:
| Condition                                              | Task Adjustment       | Explanation                                                                                             |
|--------------------------------------------------------|-----------------------|---------------------------------------------------------------------------------------------------------|
| If any reviewer approved the pull                      | +1 task per pull      | It's likely that the pull is good, just waiting on other slow reviewers to confirm.                      |
| If reviewers take longer than 24 hours                 | +1 task per pull      | Don't wait for lazy reviewers to be able to start a new task.                                            |
| If any reviewer requested changes                      | -1 task per pull      | They should focus on addressing the changes as a top priority.                                           |

_Originally posted by @0x4007 in https://github.com/ubiquibot/command-start-stop/pull/19#discussion_r1731441502_
            

For the sake of this conversation a completed PR is when the conditions are met in the above table.

In the current state the code checks if any review is approved or if there is 0 reviews but 24 hours have passed since the creation of the PR, however it doesn't check if there's any requested changes so it will count it as completed even if there 1 approve and 1 requested changes.

There is still a problem when the reviewer requests changes and the contributor resolves those changes but waits for the reviewer to make a new review so they can't start another task.
There's two possible solutions: check if the reviewer was requested more than x hours ago or check that changes have been marked as solved

**Comment by ishowvel** [comment]
Is this task up for grabs or is @jordan-ae still working on it

**Comment by jordan-ae** [comment]
Still working on it trying to get test pass. Giving me a bit of a hassle

**Comment by Keyrxng** [comment]
This has a big chance of not performing well as a worker with these suggestions because:

- for every assigned issue we need to paginate reviews (potentially 10s-100s) for each
- this feature increases that limit more and more

Perhaps moving to a dist build action should be considered?

**Comment by gentlementlegen** [comment]
@Keyrxng While I do agree that it will elongate the run time, I am not sure if we need to paginate reviews? I didn't go through the whole API but maybe we can only consider the last review? Or get a list of unresolved comments? I actually am worried more about hitting the fetch limit that the runtime duration itself (because runtime duration can be optimized, api are much harder to improve).

Moving to a dist build is totally possible, but the main problem is that even if the run could be instantaneous, we cannot skip the duration where the action run is being picked up by a gitHub node. This can sometimes take 1s, sometimes 5 minutes or more. Which is why it can be unreliable for commands that needs to be responsive.

**Comment by gentlementlegen** [comment]
@0x4007 Maybe we should remove the `x3` multiplier on the spec?

**Comment by 0x4007** [comment]
I am not sure because it's hard to compensate for research required to author good specifications but 1. We should still limit to the assignee reward amount and 2. I wish that I received the credit since I'm the original author!

**Comment by gentlementlegen** [comment]
@0x4007 There was a spec for splitting with the original post: https://github.com/ubiquity-os-marketplace/text-conversation-rewards/issues/74 so I could have been split.
